PortalFetch/crawlerDataProcessor.py

```diff
@@ -0,0 +1,106 @@
+from bs4 import BeautifulSoup
+from lxml.html import fromstring, tostring
+import json
+import collections
+
+
+class DataProcess:
+    def __getContents(self, courseList, html):
+        """
+        Get result contents from html.
+
+        Input: html is html of the courses, ulist is an empty list []
+        Output : ulist will include all the courses list in text
+        Parameters: List, String
+        Returns: None
+        """
+        soup = BeautifulSoup(html, 'lxml')
+        trs = soup.find_all('tr')
+        for tr in trs:
+            ui = []
+            for td in tr:
+                if td == '\n':
+                    continue
+                elif td.string:
+                    ui.append(str(td.string))
+                else:
+                    ui.append(td)
+            if len(ui) == 19:
+                temp = []
+                for u in ui:
+                    if u is not None:
+                        if type(u) is not str:
+                            soup = BeautifulSoup(str(u), 'lxml')
+                            u = soup.get_text()
+                        temp.append((u))
+                courseList.append(temp[:18])
+
+
+    def __deputyList(self, courseList):
+        """
+        Deputy courseList to a json file.
+
+        Input: courseLise is courses list from __getContents
+        Output : .json file  will include all the courses informatiom
+        Parameters: List,
+        Returns: None
+        """
+        dic = {}
+        title = ['Select', 'CRN', 'Coreq', 'Subj', 'Crse', 'Sec', 'Cmp', 'Cred', 'Title', 'Days', 'Time', 'Act', 'Rem',
+                 'WL Rem', 'Instructor', 'Date (MM/DD)', 'Location', 'Attribute', 'lab']
+        for i in range(len(courseList)):
+            if courseList[i][0] == 'Select':
+                subj = courseList[i+1][3]
+                d = []
+                dic[subj] = d
+            else:
+                if courseList[i][0] != '\xa0':
+                    di = {}
+                    count = -1
+                    for ele in courseList[i]:
+                        count += 1
+                        if count != 3 and count != 2 :
+                            di[title[count]] = ele if ele != '\xa0' else ""
```

Comment:
Please be consistent when choosing single quote or double quote, as suggested by [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)